From d10bfdbcc6403aff00e6e5eea8572c50eaed5538 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Tue, 10 Oct 2023 14:53:19 +0200
Subject: [PATCH 0608/1141] mtd: spi-nor: macronix: add late_init API

All capabilities from latest Macronix SPI NOR are extracted from the
SFDP table, so move the registration of spi_nor_set_4byte_addr_mode
and macronix_nor_set_octal_dtr functions in late_init API.

This patch is partially back-ported from kernel v6.6.

Change-Id: Id62d66ad05c2a664791015b9ea4055a7d025097b
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/333814
Domain-Review: Eric FOURMONT <eric.fourmont-ext@st.com>
Reviewed-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Tested-by: Eric FOURMONT <eric.fourmont-ext@st.com>
---
 drivers/mtd/spi-nor/macronix.c | 5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/mtd/spi-nor/macronix.c
+++ b/drivers/mtd/spi-nor/macronix.c
@@ -140,6 +140,10 @@ static const struct flash_info macronix_
 static void macronix_nor_default_init(struct spi_nor *nor)
 {
 	nor->params->quad_enable = spi_nor_sr1_bit6_quad_enable;
+}
+
+static void macronix_nor_late_init(struct spi_nor *nor)
+{
 	nor->params->set_4byte_addr_mode = spi_nor_set_4byte_addr_mode;
 	nor->flags &= ~SNOR_F_HAS_16BIT_SR;
 	nor->flags |= SNOR_F_HAS_LOCK;
@@ -164,6 +168,7 @@ static int macronix_nor_read_id(struct s
 
 static const struct spi_nor_fixups macronix_nor_fixups = {
 	.default_init = macronix_nor_default_init,
+	.late_init = macronix_nor_late_init,
 	.read_id = macronix_nor_read_id,
 };
 
